cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go

```diff
@@ -319,6 +319,10 @@ func (machinedeployment *MachineDeployment) DecreaseTargetSize(delta int) error
 	if err != nil {
 		return err
 	}
+	if int(size+int64(delta)) < machinedeployment.minSize {
+		klog.Warningf("Cannot go below min size= %d for machineDeployment %s, requested target size= %d . Setting target size to min size", machinedeployment.minSize, machinedeployment.Name, size+int64(delta))
+		return machinedeployment.mcmManager.SetMachineDeploymentSize(machinedeployment, int64(machinedeployment.minSize))
+	}
```

Comment:
Why do we need this change? This method is called only by `fixNodeGroupSize` logic and we won't use it with this PR.

Reply:
Yes , but for future use cases, lets say upstream start using it for some other logic, this would act as a safegaurd